File: solutions/c/grains/1/grains.c
#include "grains.h"

uint64_t square(uint8_t index) {
    if (index < 1 || index > 64)
        return 0; // или какое-то другое значение, обозначающее ошибку

    uint64_t res = 1;
    for (uint8_t i = 2; i <= index; ++i) {
        res <<= 1;
    }

    return res;
}

uint64_t total(void) {
    uint64_t sum = 0;
    uint64_t res = 1;

    for (int i = 1; i <= 64; ++i) {
        sum += res;
        res <<= 1;
    }

    return sum;
}

File: solutions/go/grains/1/grains.go
package grains

import (
    "fmt"
    "math"
)

func Square(number int) (uint64, error) {
    if number < 1 || number > 64 {
        return uint64(0), fmt.Errorf("Incorrect value")
    }
	return uint64(math.Pow(2.0, float64(number - 1))), nil
}

func Total() uint64 {
	var res uint64
    res = 0
    for i := 1; i <= 64; i++ {
        data, _ := Square(i)
        res += data
    }

    return res
}
